mock_runtime.cpp

```diff
@@ -210,7 +234,7 @@ int main(int argc, char** argv) {
     //pthread_create(&worker_thread[p], nullptr, (void *(*)(void *))lib_main, nullptr);
     // Now, we call a wrapper function and pass the main function as an argument so that we can insert a hook for enqueueing a poison pill
     // (or otherwise telling the runtime that the application is done executing)
-    pthread_create(&worker_thread[p], nullptr, (void *(*)(void *)) thread_exec_function, lib_main);
+    pthread_create(&worker_thread[p], nullptr, (void *(*)(void *)) thread_exec_function, &objCallStruct);
```

Comment:
Using the same `objCallStruct` for each instance should be fine so long as we don't have a situation where an application modifies the shared `args` (i.e. `argv`) pointer. I'm certainly not planning on having apps that mutate `argv` directly, and `argc` is passed by value anyway, so this lgtm